### PR TITLE
New package: cquery-v20180302

### PR DESCRIPTION
**Pull request by jtalowell**

### CODE REVIEW

File: srcpkgs/cquery/template [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+configure_args="--llvm-config=llvm-config"
+hostmakedepends="git"
+makedepends="clang llvm"
+depends="clang"
+short_desc="Low-latency language server for C, C++, and Objective-C"
+maintainer="jtalowell <jtalowell@protonmail.com>"
+license="MIT"
+homepage="https://github.com/cquery-project/cquery/"
+
+do_fetch() {
```
Comment (maxice8) [inline srcpkgs/cquery/template]: we don't accept that. use a tagged release

File: srcpkgs/cquery/template [hunk 2]
```diff
@@ -0,0 +1,29 @@
+pkgname="cquery"
```
Comment (maxice8) [inline srcpkgs/cquery/template]: should not be quoted

File: srcpkgs/cquery/template [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+version="v20180302"
```
Comment (maxice8) [inline srcpkgs/cquery/template]: should not be quoted

File: srcpkgs/cquery/template (hunk 2 above)
Comment (maxice8) [inline srcpkgs/cquery/template]: add the template header on top of template like so

# Template file for 'cquery'

File: srcpkgs/cquery/template (hunk 1 above)
Comment (jtalowell) [reply srcpkgs/cquery/template]: I'm using a tagged release. I checkout the tagged release in `do_fetch`: `git checkout ${version}`.

The current release won't build without error due to usage of the deprecated `-latomic` flag for `ld` so I've checked out the release using git but build using the updated build script. This won't be an issue with the next release.

Is there alternative way you recommend handling this?

File: srcpkgs/cquery/template (hunk 2 above)
Comment (jtalowell) [reply srcpkgs/cquery/template]: Fixed.

File: srcpkgs/cquery/template (hunk 1 above)
Comment (maxice8) [reply srcpkgs/cquery/template]: the way that is present on most templates

distfiles="https://github.com/cquery-project/cquery/archive/v${version}.tar.gz"
checksum=

Comment (maxice8) [reply srcpkgs/cquery/template]: > The current release won't build without error due to usage of the deprecated -latomic flag for ld so I've checked out the release using git but build using the updated build script. This won't be an issue with the next release.

add a patch to srcpkgs/cquery/patches/\<name>.patch that fixes it.

**Comment by jtalowell** [comment]
Closing.

It looks like [loguru](https://github.com/emilk/loguru), one of the modules that `cquery` depends on, uses `execinfo.h` to provide `backtrace`. This isn't something that `musl` supports, hence the build errors.

I'll lodge an issue and reopen if it support is provided. I don't understand this enough to patch it myself.

**Comment by maxice8** [comment]
we can just patch it out, please don't close if you plan to open again later, just keep it open.

**Comment by concatime** [comment]
Why isn't cquery able to load dynamically dependencies?

**Comment by concatime** [comment]
Cause it kinda sucks to specify for each a version, and to keep them updated!

**Pull request closed without merging**